lib/react/rails/railtie.rb

```diff
@@ -65,7 +65,7 @@ class Railtie < ::Rails::Railtie
       config.after_initialize do |app|
         # Server Rendering
         # Concat component_filenames together for server rendering
-        app.config.react.components_js = lambda {
+        app.config.react.components_js ||= lambda {
```

Comment:
@vipulnsward It would be much easier with a whiteboard :-) but I would appreciate if you'd explain more because I wasn't aware of this problem and would like to learn more.

 > x is not always guaranteed to be 10 here by ruby

What are the possible values then? I assumed it would either remain unchanged or be set to 10.

I still don't think either implementation is threadsafe. `Rails.app.config` is global so in theory another thread could set it to `nil` and break either case.

For example,if our react-rails init thread has just checked if `app.config.react.components_js` is `nil` either by your `.nil?` guard or the `||=` statement, and some other thread sets it to `nil` at that exact moment, we won't assign it any value and it will remain `nil` despite it looking like our intention was to assign it a default value when `nil`.

So are you specifically referring to this code being run simultaneously by multiple threads and leading to unexpected outcomes?  By my reading, the unexpected outcome would be the creation of two lambdas which is wasteful but wouldn't break any functionality. Can you explain a case where it would break?